app/src/main/java/com/android/streetworkapp/ui/map/MapUi.kt

```diff
@@ -62,6 +62,7 @@ fun MapScreen(
                 state = MarkerState(position = LatLng(park.lat, park.lon)),
                 onClick = {
                   parkViewModel.getOrCreateParkByLocation(park)
+                  parkViewModel.setParkLocation(park)
```

Comment:
Here too, using the Park MVVM to set the current park already sets the ParkLocation. Wouldn't it be better to only use the current park?

Reply:
I think it is better to separate the two, as creating the park is critical for our app, and it should rely as less as possible on other API. (Also as stated earlier, your implementation does not seem to work after a few tests)

Reply:
I think in this case it's not a question of separation, given that this line itself doesn't call the API but just sets the location, as said above it just duplicates the location. From my side removing this line and only use the current park works correctly (see comment above).